File: COMP9041/lab06/whale.py
#!/usr/bin/env python

# reimplement whale.pl using python
import sys,re

whale_record = sys.stdin.readlines();
curr_whale = sys.argv[1]
pod_count = 0
individual_count = 0

for pod in whale_record:
    pattern = "^[0-9]+ +" + curr_whale + "$"
    # print "Current pattern : " + pattern
    
    if re.match(pattern, pod):
        
        pod_count += 1
        individual_ob = pod.split(" ")[0] # note that we can not use " +" in python
        individual_count += int(individual_ob)
    
print "{} observations: {} pods, {} individuals".format(curr_whale, pod_count, individual_count)

File: COMP9041/lab06/whale.pl
#!/usr/bin/perl -w

# reads whale observations in the above format until the end-of-input 
@whale_record = ();
while($pod = <STDIN>){
    push @whale_record, $pod;
}

# check the appearance of given species
$curr_whale = $ARGV[0];
$pod_count = 0;
$individual_count = 0;
foreach $pod(@whale_record){
    if ($pod =~ m/^[0-9]+ +$curr_whale$/) {
        $pod_count++;
        @splitted = split(" +", $pod); # split by one or more spaces
        $individual_ob = $splitted[0];
        $individual_count += $individual_ob;
    }
}

print "$curr_whale observations: $pod_count pods, $individual_count individuals\n";
